src/ccmain/tesseractclass.cpp

```diff
@@ -79,6 +79,30 @@ Tesseract::Tesseract()
                  "Thresholding "
                  "method: 0 = Otsu, 1 = Adaptive Otsu, 2 = Sauvola",
                  this->params())
+    , INT_MEMBER(thresholding_sauvola_window_half_width, 25,
+                 "Window half-width for measuring local statistics",
+                 this->params())
+    , double_MEMBER(thresholding_sauvola_kfactor, 0.40,
+                 "Factor for reducing threshold due to variance. Must be 0 or positive number",
+                 this->params())
+    , INT_MEMBER(thresholding_adaptive_otsu_tile_x, 300,
+                 "Desired tile width. Actual size may vary. Must be >= 16",
+                 this->params())
+    , INT_MEMBER(thresholding_adaptive_otsu_tile_y, 300,
+                 "Desired tile height. Actual size may vary. Must be >= 16",
+                 this->params())
+    , INT_MEMBER(thresholding_adaptive_otsu_smooth_x, 0,
+                 "Half-width of convolution kernel applied to threshold array: "
+                 "Use 0 for no smoothing",
+                 this->params())
+    , INT_MEMBER(thresholding_adaptive_otsu_smooth_y, 0,
+                 "Half-width of convolution kernel applied to threshold array: "
+                 "Use 0 for no smoothing",
+                 this->params())
```

Comment:
In the interest of simplicity, let's not distinguish x and y here, only a single value for both.

Reply:
I prefer to mirror Leptonica's parameters. If Leptonica provides them both it means they can have different values.

I also tried to use similar (but not identical) names for the configurable parameters to Leptonica's variables names.

Reply:
Of course you can smooth differently in x and y direction, but this will not be useful or necessary 99% of the time.

I strongly recommend _not_ just mimicing Leptonica parameters here: this is an implementation detail. Leptonica's job is to provide an implementation for certain mathematical operations (as a backend), Tesseract's is to do practical document processing with that (as a frontend). You want to _hide_ details and complexity – this will actually give the users _more_ leverage.

Reply:
I changed `smooth_x`, `smooth_y` to just `smooth` as suggested.